### PR TITLE
Add a validator node proposal

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Fullnode got unnecessarily complex over time.

#### Summary of Changes

Add a proposal for a new simpler validator node.

### CODE REVIEW

File: book/src/validator.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+with a PoH hash. In the validator, we verify that hash, peel it off, and
+process the transactions in exactly the same way. The only difference is that
+if a validator sees a bad transaction, it can't simply boot it like the leader
+does, because that would cause the PoH hash to change.  Instead, it rejects the
+whole block. The other difference between the pipelines is what happens *after*
+banking. The leader broadcasts entries to downstream validators whereas the
+validator will have already done that in RetransmitStage, which is a
+confirmation time optimization.  The validation pipeline, on the other hand,
+has one last step. Any time it finishes processing a block, it needs to weigh
+any forks it's observing, possibly cast a vote, and if so, reset its PoH hash
```
Comment (pgarg66) [inline book/src/validator.md]: Should we show voting in the diagram? Or, is that signified by the array from Retransmit Stage to Gossip Service?

File: book/art/validator.bob [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+             |  Validator                                               |
+             |                               .----------.               |
+             |                               | Gossip   |               | .------------------.
+             |                               | Service  +---------------->| Validators       |
+             |                               `----------`               | |                  |
+             |                                  ^                       | |                  |
+             |                                  |                       | |  .------------.  |
+             |                           .------+-----.  .-----------.  | |  |            |  |
+             |                           | Retransmit |<-+ BlobFetch |<------+ Upstream   |  |
+             |                           | Stage      |  | Stage     |  | |  | Validators |  |
```
Comment (rob-solana) [inline book/art/validator.bob]: where's our favorite: replay_stage?

Comment (garious) [reply book/art/validator.bob]: Integrated into BankingStage

File: book/src/validator.md (hunk 1 above)
Comment (garious) [reply book/src/validator.md]: Done, thanks!

**Comment by garious** [comment]
@aeyakovenko, that's all part of the TPU here.

**Comment by aeyakovenko** [comment]
I want to hold of on this until after beacons

**Comment by garious** [comment]
@aeyakovenko, there's now a TPU diagram. Ready to re-review?

**Comment by aeyakovenko** [comment]
So I don't see how the following problems are solved
* broadcast stage uses the bank to generate the broadcast group
* poh recorder tracks the tpu bank and forwards it with the entries to the broadcast stage
* where does the vote happen

**Comment by garious** [comment]
> So I don't see how the following problems are solved
> 
> * broadcast stage uses the bank to generate the broadcast group
> * poh recorder tracks the tpu bank and forwards it with the entries to the broadcast stage
> * where does the vote happen

The diagrams intend to show the flow of transactions, not transfer of Banks or messages to Gossip (votes).  MulticastStage, like the current BroadcastStage, can be sent references to Banks. Likewise, no change to the voting strategy other than the TPU would push the vote via a trait instead of directly to ClusterInfo, which has I/O dependencies.

It might be best to think about this redesign as solving two distinct problems:
* The leader rotation logic is not implemented in an I/O-free component. It's a frequent pain point that would likely go away if it didn't take so much effort to encode tests for all the edge cases.
* There's too much space between the TPU and TVU, as well as BroadcastStage and RetransmitStage. I think there's a big opportunity to unify a lot of that logic.

File: book/src/validator.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+can toggle leader mode on whenever the validator's ID shows up in the
+leader schedule.
+
+<img alt="Validator block diagram" src="img/validator.svg" class="center"/>
+
+## Notable changes
+
+* No threads are shut down to switch out of leader mode. Instead, FetchStage
+  should forward transactions to the next leader.
+* FetchStage from TPU and BroadcastStage boot entirely
```
Comment (mvines) [inline book/src/validator.md]: nit:  I'd boot the term "boot" from the doc entirely.   Synonyms like "removed" are more understandable to those not familiar with gregisms.  👢

Comment (garious) [reply book/src/validator.md]: Done in 689147be9cf93333b0dc624ff1bc2782b008fe9d

**Comment by solana-grimes** [comment]
:scream: New commits were pushed while the automerge label was present.
